src/frontend/planner_test/tests/testdata/index_selection.yaml

```diff
@@ -416,15 +416,15 @@
               BatchExchange { order: [], dist: Single }
                 BatchScan { table: idx1, columns: [idx1.t1._row_id], scan_ranges: [idx1.a > Int32(1) AND idx1.a < Int32(8)], distribution: SomeShard }
               BatchExchange { order: [], dist: Single }
-                BatchScan { table: idx3, columns: [idx3.t1._row_id], scan_ranges: [idx3.c >= Int64(8) AND idx3.c <= Int64(9)], distribution: SomeShard }
+                BatchScan { table: idx3, columns: [idx3.t1._row_id], scan_ranges: [idx3.c = Int64(8) OR idx3.c = Int64(9)], distribution: SomeShard }
   batch_local_plan: |
     BatchLookupJoin { type: Inner, predicate: idx1.t1._row_id IS NOT DISTINCT FROM t1._row_id AND (((t1.a > 1:Int32) AND (t1.a < 8:Int32)) OR ((t1.c >= 8:Int32) AND (t1.c <= 9:Int32))), output: [t1.a, t1.b, t1.c, t1.p] }
       BatchHashAgg { group_key: [idx1.t1._row_id], aggs: [] }
         BatchUnion { all: true }
           BatchExchange { order: [], dist: Single }
             BatchScan { table: idx1, columns: [idx1.t1._row_id], scan_ranges: [idx1.a > Int32(1) AND idx1.a < Int32(8)], distribution: SomeShard }
           BatchExchange { order: [], dist: Single }
-            BatchScan { table: idx3, columns: [idx3.t1._row_id], scan_ranges: [idx3.c >= Int64(8) AND idx3.c <= Int64(9)], distribution: SomeShard }
+            BatchScan { table: idx3, columns: [idx3.t1._row_id], scan_ranges: [idx3.c = Int64(8) OR idx3.c = Int64(9)], distribution: SomeShard }
```

Comment:
Based on our current scheduler policy, did we schedule one task for each or condition?

Reply:
IIUC, a scan stage may include multiple scan ranges and they're joined to execute concurrently.

https://github.com/risingwavelabs/risingwave/blob/fcff8d24963f6a95086385cc289f052204afa532/src/batch/src/executor/row_seq_scan.rs#L327-L336

Reply:
For each compute node, we will merge all the range scan belonged to it into a single task.

Reply:
If mutiple scan ranges in a task are executing concurrently, seems no need to schedule one task for each scan ranges.

Reply:
> If mutiple scan ranges in a task are executing concurrently, seems no need to schedule one task for each scan ranges.

Currently we schedule one task for each PU

```rust
pub struct TableScanInfo {
    /// Indicates the table partitions to be read by scan tasks. Unnecessary partitions are already
    /// pruned.
    ///
    /// `None` if the table is not partitioned (system table).
    pub partitions: Option<HashMap<ParallelUnitId, PartitionInfo>>,
}

pub struct PartitionInfo {
    pub vnode_bitmap: Buffer,
    pub scan_ranges: Vec<ScanRangeProto>,
}
```